code/mob/living/critter/sawfly.dm

```diff
@@ -0,0 +1,206 @@
+	icon_state = "sawfly"
+	flags = TABLEPASS
+
+	var/beeptext = " "
+	var/deathtimer = 0 // for catastrophic failure on death
+	var/isnew = TRUE // for seeing whether or not they will make a new name on redeployment
+	var/sawflynames = list("A", "B", "C", "D", "E", "F", "V", "W", "X", "Y", "Z", "Alpha", "Beta", "Gamma", "Lambda", "Delta")
+	var/beeps = list('sound/machines/sawfly1.ogg','sound/machines/sawfly2.ogg','sound/machines/sawfly3.ogg') // custom noises so they cannot be mistaken for ghostdrones or borgs
+	health = 50 //this value's pretty arbitrary, since it's overridden when they get their healtholders
+	var/fliesnearby = 0 //for rolling chance to beep
+	var/friends = list()
+	misstep_chance = 40 //makes them behave more like drones, and harder to kite into a straightaway then shoot
+	var/list/dummy_params = list("icon-x" = 16, "icon-y" = 16) //for the manual attack_hand retaliation
+
+	//mob variables
+	custom_gib_handler = /proc/robogibs
+	isFlying = 1
+	can_grab = FALSE
+	can_help = FALSE
+	can_disarm = FALSE
+	use_stamina = FALSE
+	use_stunned_icon = FALSE
+	butcherable = FALSE
+	can_bleed = FALSE
+	canbegrabbed = FALSE
+	can_lie = FALSE
+	can_burn = FALSE
+	pet_text = "cuddles"
+	hand_count = 1 //stabby hands
+	setup_healths()
+		add_hh_robot(25, 1)
+		add_hh_robot_burn(25, 1)
+
+
+
+	New()
+		..()
+		if(isnew)
+			name = "Sawfly [pick(sawflynames)]-[rand(1,999)]"
+		deathtimer = rand(1, 5)
+		death_text = "[src] jutters and falls from the air, whirring to a stop."
+		beeptext = "[pick(list("beeps",  "boops", "bwoops", "bips", "bwips", "bops", "chirps", "whirrs", "pings", "purrs", "thrums"))]"
+		animate_bumble(src) // gotta get the float goin' on
+		src.set_a_intent(INTENT_HARM) // incredibly stupid way of ensuring they aren't passable but it works
+		// ai setup
+		src.mob_flags |= HEAVYWEIGHT_AI_MOB
+		src.ai = new /datum/aiHolder/sawfly(src)
+		src.is_npc = TRUE
+		START_TRACKING
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.limb = new/datum/limb/sawfly_blades
+		HH.name = "sawfly blades"
+		HH.limb_name = HH.name
+		HH.can_hold_items = FALSE
+		HH.can_range_attack = TRUE
+
+	disposing()
+		. = ..()
+		STOP_TRACKING
+
+
+	proc/foldself()
+		if(!isalive(src))
+			return 0
+		else
+			var/obj/item/old_grenade/sawfly/reused/N = new /obj/item/old_grenade/sawfly/reused(get_turf(src))
+			// pass our name and health
+			N.name = "Compact [name]"
+			N.tempname = src.name
+			N.tempdam = (50 - src.health )
+			qdel(src)
+
+
+	proc/communalbeep() // distributes the beepchance among the number of sawflies nearby
+		fliesnearby = 1 //that's you, little man! :)
+		for_by_tcl(E, /mob/living/critter/robotic/sawfly)
+			if(isalive(E))
+				src.fliesnearby += 1 //that's your buddies!
+		var/beepchance = (1 / fliesnearby) * 100 //if two sawflies, give 50% chance that any one will beep
+		if(fliesnearby<3) beepchance -=20 //heavily reduce chance of beep in swarm
```

Comment:
code seems to conflict with comment - this will decrease the beep chance unless there are at least 3 in proximity

Reply:
Swarms tend to spread out, so I didn't want to have them only ever have reduced beepchance on the rarer occasion every member of a swarm is bunched up. IMO 3 is a pretty good milestone that they're densely packed.